### ISSUE TITLE
Boost dependency removal

### ISSUE THREAD

**Issue by deleisha** [OData/odatacpp-server#5]
First of all Thanks for the odatacpp-server.

I intend to develop a light weight embeddable odata service, current plan is to use libuv-tls +odatacpp-server. 

A thought which bother now is boost dependency as this will bring in some heaviness to server.
As we go through the code, boost part seem be less and can be removed.

Will the odatacpp-server team be OKAY for the removal, If so, I can start working on that.

I should have posted in the mailing list but misses to see. Sorry for it

**Comment by xuubin** [comment]
@deleisha Yes, please do that. Actually, we also noticed this problem, but have not get resource to do the job yet. Thanks for your contribution, we really appreciate that.
